extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/internal/config/TradfriGatewayConfig.java

```diff
@@ -7,18 +7,32 @@
  */
 package org.eclipse.smarthome.binding.tradfri.internal.config;
 
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.smarthome.binding.tradfri.TradfriBindingConstants;
+
 /**
  * Configuration class for the gateway.
  *
  * @author Kai Kreuzer - Initial contribution
  */
 public class TradfriGatewayConfig {
 
-    public static final String CONFIG_HOST = "host";
-    public static final String CONFIG_PORT = "port";
-    public static final String CONFIG_CODE = "code";
-
     public String host;
     public int port = 5684; // default port
     public String code;
+    public String identity;
+    public String preSharedKey;
+    
+    public Map<String, Object> getAsMap() {
```

Comment:
remove this function please...

Reply:
Do you really think this:
```
Configuration editedConfig = editConfiguration();
editedConfig.put(TradfriBindingConstants.GATEWAY_CONFIG_HOST, configuration.code);
editedConfig.put(TradfriBindingConstants.GATEWAY_CONFIG_HOST, configuration.identity);
editedConfig.put(TradfriBindingConstants.GATEWAY_CONFIG_HOST, configuration.preSharedKey);
updateConfiguration(editedConfig);
```

looks prettier than:
```
updateConfiguration(new Configuration(configuration.getAsMap()));
```

Reply:
Yes, since `getAsMap` does not exist and you are illegally modifying the configuration instance of the Thing.